daemon/algod/api/server/v2/utils.go

```diff
@@ -178,3 +217,52 @@ func decode(handle codec.Handle, data []byte, v interface{}) error {
 	}
 	return nil
 }
+
+// Helper to convert basics.StateDelta -> *generated.StateDelta
+func stateDeltaToStateDelta(d basics.StateDelta) *generated.StateDelta {
+	if len(d) == 0 {
+		return nil
+	}
+	var delta generated.StateDelta
+	for k, v:= range d {
+		delta = append(delta, generated.EvalDeltaKeyValue{
+			Key:   base64.StdEncoding.EncodeToString([]byte(k)),
+			Value: generated.EvalDelta{
+				Action: uint64(v.Action),
+				Bytes:  strOrNil(base64.StdEncoding.EncodeToString([]byte(v.Bytes))),
+				Uint:   numOrNil(v.Uint),
+			},
+		})
+	}
+	return &delta
+}
+
+func convertToDeltas(txn node.TxnWithStatus) (*[]generated.AccountStateDelta, *generated.StateDelta) {
+	var localStateDelta *[]generated.AccountStateDelta
+	if len(txn.ApplyData.EvalDelta.LocalDeltas) > 0 {
+		d := make([]generated.AccountStateDelta, 0)
+		accounts := txn.Txn.Txn.Accounts
+
+		for k, v := range txn.ApplyData.EvalDelta.LocalDeltas {
+			// Resolve address from index
+			var addr string
+			if k == 0 {
+				addr = txn.Txn.Txn.Sender.String()
+			} else {
+				if int(k - 1) < len(accounts) {
+					addr = txn.Txn.Txn.Accounts[k-1].String()
+				} else {
+					addr = fmt.Sprintf("Invalid Address Index: %d", k - 1)
```

Comment:
I went back and forth on this. If we return an error instead of the partial result, then there is effectively no way to get transaction at all.

This is also probably an overzealous error check to begin with, since the transaction would be invalid and shouldn't have been confirmed / contain an ApplyData object if this case occurred.

Reply:
Correct - it's an indication that we have an internal error. But that doesn't mean that we should return an invalid value in the "address" field.

Reply:
And you can't just drop this entry and return the result, because the result of not be "complete".

Reply:
maybe we should log this error and return an error ?